filebeat/channel/connector.go

```diff
@@ -117,3 +107,63 @@ func (c *pipelineConnector) ConnectWith(cfg *common.Config, clientCfg beat.Clien
 	}
 	return outlet, nil
 }
+
+// buildProcessorList assembles the Processors for a pipelineConnector.
+func buildProcessorList(
+	beatInfo beat.Info, config inputOutletConfig, clientCfg beat.ClientConfig,
+) (*processors.Processors, error) {
+	procs := processors.NewList(nil)
+
+	// Processor ordering is important:
+	// 1. Index configuration
+	if !config.Index.IsEmpty() {
+		staticFields := fmtstr.FieldsForBeat(beatInfo.Beat, beatInfo.Version)
+		timestampFormat, err :=
+			fmtstr.NewTimestampFormatString(&config.Index, staticFields)
+		if err != nil {
+			return nil, err
+		}
+		indexProcessor := &addFormattedIndex{timestampFormat}
+		procs.List = append(procs.List, indexProcessor)
+	}
+
+	// 2. ClientConfig processors
+	if lst := clientCfg.Processing.Processor; lst != nil {
+		procs.List = append(procs.List, lst)
+	}
+
+	// 3. User processors
+	userProcessors, err := processors.New(config.Processors)
+	if err != nil {
+		return nil, err
+	}
+	procs.List = append(procs.List, userProcessors)
+	return procs, nil
+}
+
+//////////////////////////////
+// addFormattedIndex is a Processor to set an event's "raw-index" metadata field
+// with a given TimestampFormatString. The elasticsearch output interprets
+// that field as specifying the (raw string) index the event should be sent to;
+// in other outputs it is just included in the metadata.
+
+type addFormattedIndex struct {
```

Comment:
godoc should not have newline between the doc string and the definition.

If you feel the need for `//////////////////////` to separate sections in a file, then maybe consider a separate file.

Although you can have type definitions anywhere, the code base typically puts type definitions at the top of the file. For consistency reasons I'd like to keep this. If you still find it difficult to navigate to your code, consider a separate file.